### DISCUSSION TITLE
Can you elaborate on why this issue was locked?

### DISCUSSION THREAD

**Discussion by bkuhl** [laravel/nova-issues#6029] [General]
I noticed that https://github.com/laravel/nova-issues/issues/1003 was locked, limiting those who can contribute to it.  Can you perhaps elaborate why this was locked rather than just left in a "Closed" state?  The community can't collaborate on it anymore.

**Comment by bkuhl** [comment]
I just spent about an hour trying to get rid of "Edit" and "Edit Attached" options on a relationship, which turned out to be solved by the change in this issue, so it seems to be ongoing from my perspective.

**Reply by davidhemphill** [reply]
Looked into this. I have a `Project` resource that belongs to many `Tag`. I was able to hide the edit attached button by adding a `attachTag` method to my `ProjectPolicy` and returning `false` inside it.

Can you confirm this doesn't work for you?

**Reply by bkuhl** [reply]
Bah... could've sworn I tried this!!  Yeah, that worked for me too.  Sorry!

**Comment by bkuhl** [comment]
I'm understanding more of what's happening here within the policies (putting this here for my own reference).

If you want to disable the "Edit Attached" in the below screenshot so you have to perform the delete/attach separately is when the below `contains()` check is required.

```
    public function attachRole(User $currentUser, User $user, $role)
    {
        // Allow items not yet attached to be attachable
        if (!$user->roles->contains('id', $role->id)) {
            return true;
        }

        // Disables "Edit Attached"
        return false;
    }
```

<img width="829" alt="Screenshot 2023-12-02 at 10 01 30 AM" src="https://github.com/laravel/nova-issues/assets/524933/20167a01-a62f-4d4e-b688-5fa2cc3ccbc4">

When attaching, if the `contains()` step is not present, the `Roles` dropdown will not contain any records as it's set to `false`, so the `contains()` check ensures that on this dropdown Roles that aren't already added to the `User` will be listed.

<img width="692" alt="Screenshot 2023-12-02 at 10 06 37 AM" src="https://github.com/laravel/nova-issues/assets/524933/6ab2a166-9b1c-4d82-b50d-1dac36f82b47">

To keep things simple, you could just allow the relations to be edited, depending on the desired outcome.
